nifi-nar-bundles/nifi-poi-bundle/nifi-poi-processors/src/main/java/org/apache/nifi/processors/poi/ConvertExcelToCSVProcessor.java

```diff
@@ -202,9 +204,9 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         final CSVFormat csvFormat = CSVUtils.createCSVFormat(context);
 
         //Switch to 0 based index
-        final int firstRow = context.getProperty(ROWS_TO_SKIP).asInteger() - 1;
+        final int firstRow = Integer.parseInt(context.getProperty(ROWS_TO_SKIP).evaluateAttributeExpressions().getValue()) - 1;
```

Comment:
I'd do:
````java
final int firstRow = context.getProperty(ROWS_TO_SKIP).evaluateAttributeExpressions(flowFile).asInteger() - 1;
````
Note that I use the EL evaluation against the flow file, otherwise you won't look at the incoming FF attributes.

Reply:
Since you're on this... could also be nice to do the EL evaluation against the incoming flow file on DESIRED_SHEETS few lines above...

Reply:
Okay, will take care.

Reply:
The DESIRED_SHEETS is already supporting EL

Reply:
I got what you meant, evaluate the expression across the incoming flowfile attribute. I will include the change.